src/metagen/src/shared/files.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use std::{borrow::Cow, collections::HashMap};
+
+use crate::interlude::*;
+use common::typegraph::{
+    visitor::{Edge, PathSegment},
+    visitor2::{self, NearestFn, VisitNext},
+    Typegraph,
+};
+
+#[derive(Debug)]
+pub struct TypePath(pub Vec<Cow<'static, str>>);
```

Comment:
Could you make this non Rust specific and use types instead like before in order to share it between implementations.